### PR TITLE
Phase folded plot in SourceTable

### PR DESCRIPTION
**Pull request by mcoughlin**
This PR adds a phase folded plot to the SourceTable page. 

Questions for @guynir42 I have are:
(i) Is this what you envisioned in the handler?
(ii) I added a "period_exists" flag. What do you think?

### CODE REVIEW

File: skyportal/handlers/api/photometry.py [hunk 1]
```diff
@@ -1062,16 +1063,37 @@ def delete(self, photometry_id):
...
+            period = None
+            annotations = (
+                Annotation.query_records_accessible_by(self.current_user)
+                .filter(Annotation.obj_id == obj_id)
+                .all()
+            )
+            for an in annotations:
+                if not isinstance(an.data, dict):
+                    continue
+                if 'period' in an.data:
```
Comment (guynir42) [inline skyportal/handlers/api/photometry.py:1088]: What about capital letters, e.g., `"Period"` as the key?

File: static/js/components/VegaFoldedPlot.jsx [hunk 2]
```diff
@@ -0,0 +1,207 @@
...
+        },
+        color: {
+          field: "filter",
+          type: "nominal",
+        },
+        tooltip: [
+          { field: "magAndErr", title: "mag", type: "nominal" },
+          { field: "filter", type: "ordinal" },
+          { field: "mjd", type: "quantitative" },
+          { field: "daysAgo", type: "quantitative" },
```
Comment (guynir42) [inline static/js/components/VegaFoldedPlot.jsx:73]: Replace `daysAgo` with `phase`?

**Review by guynir42 — COMMENTED** [review]
This actually looks really good already. There are some minor comments below. 

Also, we should probably document this feature, explaining how to add a period to a source and under what conditions would the phase plot appear. My first instinct is to put it into `advanced_usage.md` along with the HR diagram documentation, but maybe there's a better place. Maybe there should be a dedicated doc file for periodic sources, which will be a good place to explain about the period finding page etc. (you don't have to write that documentation, only the part about the annotations and the phase plot).

File: skyportal/handlers/api/photometry.py [hunk 3]
```diff
@@ -1062,16 +1063,37 @@ def delete(self, photometry_id):
 class ObjPhotometryHandler(BaseHandler):
     @auth_or_token
     def get(self, obj_id):
+        phase_fold_data = self.get_query_argument("phaseFoldData", False)
```
[the remark this answers is not in the document]
Comment (mcoughlin) [reply skyportal/handlers/api/photometry.py]: oops

File: skyportal/handlers/api/photometry.py [hunk 4]
```diff
@@ -1062,16 +1063,37 @@ def delete(self, photometry_id):
...
+            annotations = (
+                Annotation.query_records_accessible_by(self.current_user)
+                .filter(Annotation.obj_id == obj_id)
+                .all()
+            )
+            for an in annotations:
+                if not isinstance(an.data, dict):
+                    continue
+                if 'period' in an.data:
+                    period = an.data['period']
```
[the remark this answers is not in the document]
Comment (mcoughlin) [reply skyportal/handlers/api/photometry.py]: added. see what you think.

File: skyportal/handlers/api/source.py [hunk 5]
```diff
@@ -630,6 +637,7 @@ def get(self, obj_id=None):
         include_spectrum_exists = self.get_query_argument(
             "includeSpectrumExists", False
         )
+        include_period_exists = self.get_query_argument("includePeriodExists", False)
```
[the remark this answers is not in the document]
Comment (mcoughlin) [reply skyportal/handlers/api/source.py]: fixed.

File: skyportal/handlers/api/photometry.py (hunk 1 above)
Comment (mcoughlin) [reply skyportal/handlers/api/photometry.py]: modified, see what you think.

File: static/js/components/VegaFoldedPlot.jsx (hunk 2 above)
Comment (mcoughlin) [reply static/js/components/VegaFoldedPlot.jsx]: done.

**Comment by mcoughlin** [comment]
@guynir42 I started that page, see what you think?

File: doc/period.md [hunk 6]
```diff
@@ -0,0 +1,7 @@
+# Periodic Timeseries Features
+
+Skyportal supports periodicity features. In particular, we provide a `Periodogram Analysis` button available from the `Source` page. This page allows for a dynamic, slider based period assignment and display.
+
+![Periodogram page](images/periodogram.png)
+
+The periods are saved as `Annotation`s, and the app searches for annotated periods with the keyword "period." For those sources with this annotation, the `SourceTable` includes a phase-folded light curve on that period.
```
Comment (guynir42) [inline doc/period.md:7]: "To add a period, post an annotation with `data: {'period': <value>}` as an annotation."
"The keyword for period can be capitalized or not." 
"If multiple annotations with periods are available, then..."
"Periods are always represented in units of days."

File: skyportal/handlers/api/photometry.py [hunk 7]
```diff
@@ -1076,18 +1080,20 @@ def get(self, obj_id):
...
             for an in annotations:
                 if not isinstance(an.data, dict):
                     continue
-                if 'period' in an.data:
-                    period = an.data['period']
-                    break
+                for period_str in period_str_options:
+                    if period_str in an.data and arrow.get(an.modified) > modified:
+                        period = an.data[period_str]
+                        modified = arrow.get(an.modified)
```
Comment (guynir42) [inline skyportal/handlers/api/photometry.py:1096]: Nice.

**Review by guynir42 — COMMENTED** [review]
As usual, the code looks good to merge, and the documentation is found lacking.

**Comment by mcoughlin** [comment]
The sass on this one.

**Review by guynir42 — APPROVED** [review]

**Pull request merged**